src/apps/vpn/controller.cpp

```diff
@@ -582,10 +602,21 @@ void Controller::handshakeTimeout() {
   serverUnavailable();
 }
 
+void Controller::vpnSessionPingTimeout() {
```

Comment:
Not sure this code should live in the controller, but instead in Telemetry. I'm really not sure though, so please argue with me :D

Reply:
I can see that. We can really move all of this logic to the `telemetry` class and start/end timers or submit pings on `controller::stateChanged()`